immich-app/immich
# refactor(server)!: add/remove album assets

### Breaking Change:  Mobile app and server must both be up to date

If the mobile app and server are out of sync the following will no longer work:
- Add assets to albums
- Remove assets from albums

<hr>

TODO:
- [x] Web testing
- [x] Mobile testing
- [ ] Perf testing
- [x] Access testing
- [x] Test removing current thumbnail

Test Scenarios
- Add own assets to album
- Remove own assets from album
- Add own assets to a non-owned, shared album
- Drag and drop new assets to a non-owned, shared album (lots of notifications)
- Remove own assets from a non-owned, shared album (new feature :tada:)
- Remove a partner shared asset from a non-owned, shared album (via API confirmed no remove access without partner sharing and confirmed removed success with partner sharing)
- Remove current thumbnail asset from album and verify new thumbnail is set
- Remove last asset from album and verify album has null thumbnail

----
Mobile Testing

- Add asset(s) from album viewer (owned/shared)
- Add asset(s) from the timeline (owned/shared)
- Add asset from the detail viewer (owned/shared)
- Remove asset(s) from the album viewer (owned)
- Add asset(s) from the web and check the presence on the mobile app
